apps/sms/js/activity_handler.js

```diff
@@ -281,7 +280,7 @@ var ActivityHandler = {
 
   // Launch the UI properly
   launchComposer: function ah_launchComposer(activity) {
-    Navigation.toPanel('composer', { activity: activity });
+    return Navigation.toPanel('composer', { activity: activity });
```

Comment:
be careful as you really change the semantic here; before we were returning right away, now we still return right away :) but if we use the promise we'll wait until the end of the transition, which takes time. So yeah, I'm not against this, but just be careful.